### PR TITLE
DMP-4024: Update endpoints to GET events to add is_data_anonymised

### PR DESCRIPTION
**Pull request by Ben-Edwards-cgi**
### Change description ###
# Git Diff Summary

This Git diff introduces changes across several files, primarily focusing on the addition of an `is_data_anonymised` field to event-related data structures, tests, and OpenAPI specifications. It also includes minor refactoring and adjustments to formatting.

## Highlights

- **Added `is_data_anonymised` Field**:
  - New field added to JSON responses in `CaseControllerGetEventByCaseIdTest` and `HearingsGetEventsControllerTest`.
  - Reflected in OpenAPI specifications for cases and hearings.

- **Event Mapping Updates**:
  - The `EventMapper` and `GetEventsResponseMapper` classes have been updated to set the new `is_data_anonymised` property based on the corresponding entity's value.

- **Test Cases Enhancements**:
  - New test cases in `EventMapperTest` and `GetEventsResponseMapperTest` to validate the correct mapping of the `is_data_anonymised` field.
  - Adjustments made to existing test expected results to include the `is_data_anonymised` field.

- **Code Refactoring**:
  - Minor refactoring for consistency, such as changing `String endpointUrl` to `String endpointUrl` as a `final` variable.
  - Formatting improvements in various JSON structures for better readability.

- **New Expected Response Files**:
  - Added expected response files for tests to account for the new `is_data_anonymised` field, including variations for anonymized and non-anonymized events.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/darts/cases/mapper/EventMapperTest.java [hunk 1]
```diff
@@ -30,8 +32,28 @@ void testMapGetEventsByCaseId() {
...
+        eventEntity.setDataAnonymised(true);
+        List<EventEntity> eventEntityList = Lists.newArrayList(eventEntity);
+        List<Event> events = EventMapper.mapToEvents(eventEntityList);
+        Event event = events.get(0);
+        assertEquals(1, event.getId());
+        assertEquals(102, event.getHearingId());
+        assertEquals(hearingDate.toLocalDate(), event.getHearingDate());
+        assertEquals("eventName", event.getName());
+        assertEquals("event text", event.getText());
+        assertEquals(hearingDate, event.getTimestamp());
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/darts/cases/mapper/EventMapperTest.java:52]: minor: 
these verifications are already taken care of by another test we have on this class `testMapGetEventsByCaseId`. I think we can remove them and leave only line 53 which is the focus of this test

**Review by mario-paniccia — APPROVED** [review]
approved with one minor comment if you want

File: src/test/java/uk/gov/hmcts/darts/cases/mapper/EventMapperTest.java (hunk 1 above)
Comment (Ben-Edwards-cgi) [reply src/test/java/uk/gov/hmcts/darts/cases/mapper/EventMapperTest.java]: I have converted these tests to use ParameterizedTest to avoid duplication

**Review by davet1985 — COMMENTED** [review]
Are there any functional tests which hit these endpoints? If so, can you update them as well please.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**